One of the test cases of this test checks if ppoll is only defined
when poll.h is included with -D_GNU_SOURCE. The generated C code
for this test must be compiled with -std=c99 or newer, as otherwise,
the ppoll function will be implicitly declared to be `int ppoll()`
even if poll.h doesn't define it, thereby causing the test case
to fail.

The test failure message is:

c compiler: has function ppoll: YES
c compiler: header poll.h has symbol ppoll: YES
err ppoll should not be accessible without _GNU_SOURCE
tests/project/common/103 has header symbol/meson.build:49:5: error in function assert()
 49 |     assert(
          ^

diff -upr "muon-0.2.0.orig/tests/project/common/103 has header symbol/meson.build" "muon-0.2.0/tests/project/common/103 has header symbol/meson.build"
--- "muon-0.2.0.orig/tests/project/common/103 has header symbol/meson.build"	2024-03-27 22:11:49.923516032 +0100
+++ "muon-0.2.0/tests/project/common/103 has header symbol/meson.build"	2024-03-27 22:12:09.156904627 +0100
@@ -2,7 +2,7 @@ project(
     'has header symbol',
     'c',
     'cpp',
-    default_options: ['cpp_std=c++11'],
+    default_options: ['c_std=c99', 'cpp_std=c++11'],
 )
 
 cc = meson.get_compiler('c')
